File: src/main/java/org/apache/prepbuddy/analyzers/FileType.java
package org.apache.prepbuddy.analyzers;


import org.apache.commons.lang.StringUtils;

/**
 * File formats that are supported by TransformableRDD
 */
public enum FileType {
    CSV {
        @Override
        public String[] parseRecord(String record) {
            String[] recordAsArray = record.split(",", -1);
            return trimAll(recordAsArray);
        }

        @Override
        public String join(String[] record) {
            return StringUtils.join(record, ",");
        }

        @Override
        public String appendDelimiter(String row) {
            return row + ",";
        }
    },

    TSV {
        @Override
        public String[] parseRecord(String record) {
            String[] recordAsArray = record.split("\t", -1);
            return trimAll(recordAsArray);
        }

        @Override
        public String join(String[] record) {
            return StringUtils.join(record, "\t");
        }

        @Override
        public String appendDelimiter(String row) {
            return row + "\t";
        }
    };

    String[] trimAll(String[] record) {
        for (int i = 0; i < record.length; i++) {
            String each = record[i];
            record[i] = each.trim();
        }
        return record;
    }

    public abstract String[] parseRecord(String record);

    public abstract String join(String[] record);

    public abstract String appendDelimiter(String row);

}

File: src/main/scala/org/apache/datacommons/prepbuddy/types/FileType.scala
package org.apache.datacommons.prepbuddy.types

import org.apache.datacommons.prepbuddy.utils.RowRecord

/**
  * File formats that are supported by TransformableRDD
  */
class FileType(delimiter: String) extends Serializable {
    def appendDelimiter(row: String): String = row + delimiter

    def join(record: RowRecord): String = record.mkString(delimiter)

    def parse(record: String): RowRecord = new RowRecord(record.split(delimiter, -1).map(_.trim))
}

object CSV extends FileType(",")

object TSV extends FileType("\t")

